middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { decrypt } from '@/lib/jwt'
+import type { NextRequest } from 'next/server'
+import { NextResponse } from 'next/server'
+
+export async function middleware(request: NextRequest) {
+  const session = request.cookies.get('session')?.value
+  try {
+    // Verify JWT token
+    const payload = await decrypt(session)
+    if (!payload) {
+      // Redirect to home page if no valid session
+      return NextResponse.redirect(new URL('/', request.url))
+    }
+
+    // Valid session, continue to protected route
+    return NextResponse.next()
+  } catch (error) {
+    console.error('Middleware auth error:', error)
+    // Redirect to home page on auth error
+    return NextResponse.redirect(new URL('/', request.url))
+  }
```

Comment:
```suggestion
  try {
    const payload = await decrypt(session)

    if (!payload) {
      throw new Error('Invalid session payload')
    }
  } catch (error) {
    console.error('Middleware auth error:', error)
    return NextResponse.redirect(new URL('/', request.url))
  }

  return NextResponse.next()
```

You could also add a server function i.e. `getSession`:

```ts
export async function getSession() {
  const session = request.cookies.get('session')?.value
  try {
    return decrypt(session)
  } catch() {
    console.error('error')
  }
}
```

and then use it in middleware:

```ts
export async function middleware(request: NextRequest) {
  if(await getSession()) {
    return NextResponse.next()
  }
  return NextResponse.redirect(new URL('/', request.url))
}
```